### DISCUSSION TITLE
Increase contour line density

### DISCUSSION THREAD

**Discussion by petarlishov** [organicmaps/organicmaps#1158] [Ideas]
Currently the contour lines appear at 50m elevation intervals in some locations and 20m in others. It would be much more useful to get that data on the map at 10 meter intervals everywhere at a sufficiently high zoom level. I hope that the code and data are already there
This can be a quick solution for an alternative to the hillshading feature request (I'd argue it is a better and higher priority feature)

Examples are:
* Scotland is at 50m intervals
* Madeira is at 20m intervals

**Comment by biodranik** [comment]
There is a big disadvantage of decreasing the interval: increased file size. Now isolines are turned off for some regions because they take too much space. We may improve the compression in the future, or maybe split isolines from the main map and allow to download them separately.

**Comment by MikeEA0** [comment]
In Croatia near the sea, I first thought this feature would not work at all, because I had to zoom-in really a lot in order to make the contour lines appear...  And then I saw initially only contour lines in the sea (I am wondering why), but not at land (only when looking really closely, I saw very few ones on land). I wanted to go bicycling spontaneously, and to choose a route that has less elevation... but that didn't work out at all.

Also in South Germany, it feels like the contour lines are basically useless, because they are not fine grained enough.

> maybe split isolines from the main map and allow to download them separately.

This could also reduce overall amount of data traffic to you servers, because a decent amount of people might not be interested in contour lines at all, I would guess.

**Comment by pashky** [comment]
Yes please please can this be reconsidered. Here’s comparison of the same area with your, ahem, competitor:

![IMG_2266](https://github.com/organicmaps/organicmaps/assets/663876/3d20b26a-f8ac-4a91-9cd5-b0a18f4a578b)
![IMG_2267](https://github.com/organicmaps/organicmaps/assets/663876/8d685d08-67ff-4a4e-bdbc-8c3aec3fcdf3)

It’s not exactly Alps we’ve got here in England, but it is what is - still some terrain to be considered when planning a hike or a ride, if I didn’t know this place very well I’d be in for a nasty surprise.

That being said, aftermentioned competitor managed to gradually screw up some core functionality, so here I am, where it all started back at the mercy of MWM team. So contours or not thank you guys for still being there for us!

**Reply by biodranik** [reply]
There is a technical issue with an increased isolines size when their density increases. It is the only blocker to show isolines more often. To resolve it, isolines/3D terrain should be refactored, or, alternatively, isolines should be downloaded separately from other maps.

**Comment by Gerode** [comment]
I was hoping to switch to this app for hiking and climbing, but 50m was too large of a contour interval to be useful on my last hike in the Western US - there was a decent-sized stream valley that didn't show on my map at all, and the lake I was hiking to turned out to be quite a bit higher than the contours suggested - a steep extra 40m and then flat at the top 😛 I'll need to find another app with a finer-grained contour interval.

**Comment by vi-isnardi** [comment]
Please please please this feature is needed! This could be the greatest outdoor map app but 50m between contour line is treacherous for navigation in the backcountry...

**Comment by Rob-from-VI** [comment]
I would definitely use that feature, as I would want dense countours for local areas where I live, hike, and bike.

On the other hand I have many other maps downloaded for routing and simple viewing of locations where I do not need countour lines at all.

**Comment by xc-racer3** [comment]
I would also really, really appreciate having improved contour line density.  In British Columbia, Canada some map download areas are very large, resulting in completely unusable 500m contours.  Even just having smaller download areas and getting those areas to 20m or 50m topos would be a big improvement, although separate download selection of contours as discussed above would be even better.

**Comment by maelmadon** [comment]
Finding a solution for this issue would be very appreciated. For example, the map "Peru - North" has only 500m isolines, making it virtually unusable for hike planning...
I support the suggestion of @biodranik to make isoline available as a separate download for those who need them (solution adopted by OsmAnd~).
As a more short term solution, what about proposing two different downloads for the same area, e.g. "Peru - North iso500" and "Peru - North iso10"?
Or splitting the large areas in smaller chunks, as it is done in other regions. For example, France has 94 subregions, against only 3 subregions for the much wider country of Peru.

**Comment by pittbrookemusic-svg** [comment]
To quote Monty Python, "What I wouldn't give for 50m contour lines!"  I just downloaded Organic Maps, and in my area (British Columbia), it's only showing 500m contour lines.  Am I doing something wrong?

**Reply by biodranik** [reply]
10m contour lines with meters/feet support are coming soon, please stay tuned. A beta version also will be released with their support.
